packages/common/src/discussions/api/types.ts

```diff
@@ -520,7 +510,7 @@ export interface ICreateChannelPost extends IPostOptions, ICreateChannel {}
  * @extends {IHubRequestOptions}
  */
 export interface ICreatePostParams extends IDiscussionsRequestOptions {
-  data: ICreatePost | ICreateChannelPost;
+  data: ICreatePost;
```

Comment:
This will require changes to the post-editor component as V2 does not support creating channels on the fly while creating a post; a channelId is now required to be sent through. Need designs from Sam.